TOOLS/dylib_unhell.py

```diff
@@ -41,83 +41,106 @@ def otool(objfile, rapths):
 
 def get_rapths(objfile):
     rpaths = []
-    command = "otool -l '%s' | grep -A2 LC_RPATH | grep path" % objfile
-    pathRe = re.compile(r"^\s*path (.*) \(offset \d*\)$")
+    command = f"otool -l '{objfile}' | grep -A2 LC_RPATH | grep path"
+    path_re = re.compile(r"^\s*path (.*) \(offset \d*\)$")
 
     try:
-        result = subprocess.check_output(command, shell = True, universal_newlines=True)
-    except:
+        result = subprocess.check_output(command, shell=True, universal_newlines=True)
+    except Exception:
         return rpaths
 
     for line in result.splitlines():
-        line_clean = pathRe.search(line).group(1).strip()
+        line_clean = path_re.search(line).group(1).strip()
         # resolve @loader_path
-        if line_clean.startswith('@loader_path/'):
-            line_clean = line_clean[len('@loader_path/'):]
-            line_clean = os.path.normpath(os.path.join(os.path.dirname(objfile), line_clean))
+        if line_clean.startswith("@loader_path/"):
+            line_clean = line_clean[len("@loader_path/"):]
+            line_clean = os.path.join(os.path.dirname(objfile), line_clean)
+            line_clean = os.path.normpath(line_clean)
         rpaths.append(line_clean)
 
     return rpaths
 
 def get_rpaths_dev_tools(binary):
-    command = "otool -l '%s' | grep -A2 LC_RPATH | grep path | grep \"Xcode\\|CommandLineTools\"" % binary
+    command = (
+        f"otool -l '{binary}' | grep -A2 LC_RPATH | grep path | "
+        'grep "Xcode\\|CommandLineTools"'
+    )
     result  = subprocess.check_output(command, shell = True, universal_newlines=True)
-    pathRe = re.compile(r"^\s*path (.*) \(offset \d*\)$")
-    output = []
-
-    for line in result.splitlines():
-        output.append(pathRe.search(line).group(1).strip())
-
-    return output
+    path_re = re.compile(r"^\s*path (.*) \(offset \d*\)$")
+    return [
+        path_re.search(line).group(1).strip()
+        for line in result.splitlines()
+    ]
```

Comment:
this was also done in `ci/lint: replace format strings`. is this some personal preference since it doesn't really correspond to the commit message.

Reply:
> on a side note, i find this syntax horrifying, reversing the execution flow.

This is just  list comprehension syntax in python.

Reply:
This was a recommend by one of the linter checks, since it's faster for python to calculate. I've updated the commit message to reflect the check that was enabled at that stage.